react/features/local-recording/reducer.js

```diff
@@ -0,0 +1,46 @@
+/* @flow */
+
+import { ReducerRegistry } from '../base/redux';
+import {
+    LOCAL_RECORDING_ENGAGED,
+    LOCAL_RECORDING_STATS_UPDATE,
+    LOCAL_RECORDING_TOGGLE_DIALOG,
+    LOCAL_RECORDING_UNENGAGED
+} from './actionTypes';
+import { recordingController } from './controller';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+ReducerRegistry.register('features/local-recording', (state = {}, action) => {
+    logger.debug(`Redux state (features/local-recording):\n ${
+        JSON.stringify(state)}`);
+    switch (action.type) {
+    case LOCAL_RECORDING_ENGAGED: {
+        return {
+            ...state,
+            isEngaged: true,
+            recordingStartedAt: new Date(Date.now()),
```

Comment:
The idea of a reducer is that it is a pure function, meaning a certain set of arguments will always return a certain result. By introducing Date.now() here that's introducing something less pure, less predictable. Moving it out into action or passing it into the action would be preferred to maintain that pureness.

Mayne even the recordingController could also be passed in, but I'm torn on that.

Reply:
ohhh.. Yes it should be passed in as part of the event.